app/models/room.rb

```diff
@@ -4,5 +4,7 @@ class Room < ApplicationRecord
   has_many :receipts, through: :receipt_details
   has_many_attached :images
 
-  scope :latest, ->{order created_at: :desc}
+  scope :latest, ->{order name: :desc}
+  scope :search_furnitures, ->(key){where "furnitures.name LIKE '%#{key}%'"}
+  scope :search_rooms, ->(key){where "rooms.name LIKE '%#{key}%'"}
```

Comment:
```suggestion
  scope :search_rooms, ->(key){where "name LIKE ?", "%#{key}%"}
```
ri nhé, vì để thế dễ bị tấn công `SQL Injection`

Reply:
`"rooms.name "` cái chỗ này vì em nói với bảng furnitures, mà cả 2 bảng đều có thuộc tính name nên em ghi vào, không là nó không chạy được câu sql ạ

Reply:
`rooms.name ` chỗ này em nối với cái bảng furnitures, mà cả 2 cái ni đều có thuộc tính name, nên em bỏ vậy để cái sql nó nhận, em thử xóa thì nó không chạy được